text/0089-fee-elimination.md

```diff
@@ -0,0 +1,83 @@
+
+To address both the UX issue and the security issue with refunds, we propose a re-implementation of the refund feature, introducing delayed refunds with a percentage of "used" token made available to the user on a linear schedule. Unlike the current refund feature, the new system will functionally refund all token and not just token paid for overclaimed gas.
+
+Note that we use the term "refund" for simplicity and to emphasize the viewpoint of the user. However, token "spent" under the new system will not be transferred or otherwise associated with any account other than that of the user.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+### System view
+
+With the implementation of this proposal, execution of a deploy will still trigger the code pathways that calculate expended gas with a given gas-token conversion rate to produce a "fee." However, no transfer to the validator will take place, leaving the token that would have been expended by the user still associated with their account, but under a temporary hold. The "used" token is released for further use on a linear schedule as a function of block time.
+
+### User view
+
+From the user's perspective, all token that would, under the current system, be paid to the validator are eventually "refunded." Token are made available on a linear schedule as a function of block time.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+### Account level lazy tracking
+
+The logical account object records the "spend" on execution from deploys signed by the account's private key on a per-purse
+basis. Whenever token from some purse is to be spent in any manner, a balance check is performed, taking into account 
+
+- unencumbered purse token
+- the individual purse's holds
+- block time (which is known to the execution engine)
+
+The available balance consists of 
+
+- unencumbered purse token 
+- token from expired holds
+- token from continuing holds subject to a linear release schedule parametrized in the chainspec
+
+Optimization is left as an implementation detail, e.g., due to the inexorable march of time, all the holds' expiration times are naturally pre-sorted, enabling a quick disposal of old holds in the balance check calculation.
+
+### Changes to gas fee logic
+
+No changes, at least due to the introduction of this feature, are expected to be made to the calculation of gas fees.
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+Elimination of user fees paid to validators creates adverse incentives for validators. In the interest of keeping
```

Comment:
This needs to be broken out into a separate CEP now that we have the validator credit feature, potentially